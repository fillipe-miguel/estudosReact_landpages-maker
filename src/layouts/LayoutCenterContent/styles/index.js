import styled, { css } from 'styled-components';

export const LayoutCenterContentRoot = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const InnerHtml = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.large};
    text-align: justify;

    & h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      text-align: center;
    }
  `}
`;

export default LayoutCenterContentRoot;
