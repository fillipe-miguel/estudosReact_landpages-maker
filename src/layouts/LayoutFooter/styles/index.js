import styled, { css } from 'styled-components';

export const LayoutFooterRoot = styled.footer`
  ${({ theme }) => css`
    height: 8rem;
    text-align: center;
    font-size: ${theme.fonts.sizes.xsmall};
    background-color: ${theme.colors.white};

    a {
      pointer-events: all !important;
      text-decoration: underline !important;
    }
  `}
`;

export default LayoutFooterRoot;
