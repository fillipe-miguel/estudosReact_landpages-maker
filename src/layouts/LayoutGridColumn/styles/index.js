import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const LayoutGridColumnRoot = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;

    gap: ${theme.spacings.xxlarge};

    & ${HeadingStyles.HeadingRoot} {
      margin-bottom: ${theme.spacings.small};
    }
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      flex-direction: column;

      & ${HeadingStyles.HeadingRoot} {
        margin-bottom: ${theme.spacings.xlarge};
      }
    }
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    max-width: 35rem;
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      min-width: auto;
      text-align: center;
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css`
    max-width: 64rem;
  `}
`;

export const Image = styled.img`
  ${() => css`
    max-width: 100%;
  `}
`;
