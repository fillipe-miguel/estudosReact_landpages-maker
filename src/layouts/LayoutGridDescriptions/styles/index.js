import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    & ${HeadingStyles.HeadingRoot} {
      position: relative;
      left: 5rem;
      margin: ${theme.spacings.medium} 0;
    }

    & ${HeadingStyles.HeadingRoot}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xxxlarge};
      left: -5rem;
      top: -2.5rem;
      transform: skew(-15deg);
    }
  `}
`;
