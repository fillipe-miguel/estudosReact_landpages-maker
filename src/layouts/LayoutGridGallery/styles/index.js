import styled, { css } from 'styled-components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(270px, 1fr));
    gap: ${theme.spacings.medium};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
    height: fit-content;

    &:hover > img {
      transition: all 0.3s ease-in-out;
      transform: scale(1.2) rotate(9deg);
    }

    img {
      pointer-events: none;
      max-width: 100%;
    }
  `}
`;
