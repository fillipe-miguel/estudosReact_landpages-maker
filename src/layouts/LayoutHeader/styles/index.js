import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';
import { SectionContainerStyles } from '../../../sections';

export const LayoutHeaderRoot = styled.header`
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;

  & > ${SectionContainerStyles.SectionContainerRoot} {
    padding-top: ${({theme}) => theme.spacings.small};
    padding-bottom: ${({theme}) => theme.spacings.small};;
  }

  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-bottom: 1px solid ${theme.colors.mediumGray};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      bottom: 0;
      height: 100vh;
      visibility: ${props.visible ? 'visibility' : 'hidden'};
      opacity: ${props.visible ? '1' : '0'};
      transition: all 0.3s ease-in-out;

      & > ${SectionContainerStyles.SectionContainerRoot} {
        display: grid;
        align-items: center;

        height: 100%;
        margin: 0;

        overflow-y: auto;
      }

      & ${HeadingStyles.HeadingRoot} {
        padding-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      flex-direction: column;
      text-align: center;
      padding: ${theme.spacings.xxlarge};
    }
  `}
`;

export const ToggleMenuButton = styled.button`
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;

  display: none;

  font-size: 2.5rem;
  padding: 0.5rem;

  cursor: pointer;
  border: none;

  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      align-items: center;
    }
  `}
`;

export default LayoutHeaderRoot;
