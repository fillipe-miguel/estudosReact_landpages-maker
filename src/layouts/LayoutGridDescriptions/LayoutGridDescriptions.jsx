import P from 'prop-types';
import React from 'react';

import * as Styled from './styles';

import { Heading, TextComponent } from '../../components';
import { SectionBackground } from '../../sections';

function LayoutGridDescriptions({ title, description, setBackground, grid }) {
  return (
    <SectionBackground setBackground={setBackground}>
      <Styled.HeaderContainer>
        <Heading
          uppercase
          size="xxlarge"
          level="2"
          colorDarker={!setBackground}
        >
          {title}
        </Heading>
        <TextComponent colorDarker={!setBackground}>
          {description}
        </TextComponent>
      </Styled.HeaderContainer>
      <Styled.Grid>
        {grid.map((element) => (
          <Styled.GridElement key={element.title}>
            <Heading size="medium" level="3" colorDarker={!setBackground}>
              {element.title}
            </Heading>
            <TextComponent colorDarker={!setBackground}>
              {element.description}
            </TextComponent>
          </Styled.GridElement>
        ))}
      </Styled.Grid>
    </SectionBackground>
  );
}

LayoutGridDescriptions.defaultProps = {
  setBackground: false,
};

LayoutGridDescriptions.propTypes = {
  title: P.string.isRequired,
  description: P.string.isRequired,
  setBackground: P.bool,
  grid: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      description: P.string.isRequired,
    }),
  ).isRequired,
};

export default LayoutGridDescriptions;
