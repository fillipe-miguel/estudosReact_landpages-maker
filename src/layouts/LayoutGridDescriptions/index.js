import LayoutGridDescriptions from './LayoutGridDescriptions';

export default LayoutGridDescriptions;
