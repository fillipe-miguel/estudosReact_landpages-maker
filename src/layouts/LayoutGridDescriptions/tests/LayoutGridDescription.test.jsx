/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import LayoutGridDescriptions from '..';

import * as mocks from '../../../mocks';
import { MyThemeProvider, theme } from '../../../styles';

describe('<LayoutGridDescriptions/>', () => {
  // ================================= Teste 01 =================================
  it('Should render default', () => {
    render(
      <MyThemeProvider>
        <LayoutGridDescriptions {...mocks.layouts.gridDescription} />
      </MyThemeProvider>,
    );

    // Getting a random thing to comprove rendering
    expect(screen.queryAllByRole('heading')).toHaveLength(4);
  });

  // ================================= Teste 02 =================================
  it('Should render with background set', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutGridDescriptions
          {...mocks.layouts.gridDescription}
          setBackground
        />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toHaveStyle({
      'background-color': theme.colors.primaryColor,
    });
  });

  // ================================= Snapshot =================================
  it('Should render with background set', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutGridDescriptions
          {...mocks.layouts.gridDescription}
          setBackground
        />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
