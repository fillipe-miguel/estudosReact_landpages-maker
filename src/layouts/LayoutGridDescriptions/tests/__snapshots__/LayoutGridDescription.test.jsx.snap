// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutGridDescriptions/> Should render with background set 1`] = `
.c4 {
  color: #FFF;
  text-transform: uppercase;
  margin-bottom: 3.2rem;
  font-size: 4.8rem;
}

.c8 {
  color: #FFF;
  text-transform: none;
  margin-bottom: 3.2rem;
  font-size: 2.4rem;
}

.c5 {
  font-size: 1.6rem;
  color: #FFF;
}

.c5 p {
  margin: 2.4rem 0;
}

.c5 a {
  font-weight: bolder;
  color: #DC143C;
}

.c5 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 h1,
.c5 h2,
.c5 h3,
.c5 h4,
.c5 h5,
.c5 h6 {
  margin: 1.6rem 0;
}

.c5 ol,
.c5 ul {
  margin: 1.6rem;
}

.c5 ol > li,
.c5 ul > li {
  margin: 1.6rem 0;
  font-style: italic;
}

.c2 {
  margin-bottom: 4.8rem;
}

.c6 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(270px,1fr));
  gap: 3.2rem;
}

.c7 .c3 {
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  left: 5rem;
  margin: 2.4rem 0;
}

.c7 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 6.4rem;
  left: -5rem;
  top: -2.5rem;
  -webkit-transform: skew(-15deg);
  -ms-transform: skew(-15deg);
  transform: skew(-15deg);
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFF;
  color: #0A1128;
  background-color: #0A1128;
  color: #FFF;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.8rem;
  }
}

@media (max-width:768px) {
  .c8 {
    font-size: 2.4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    text-align: center;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c7 .c3 {
    left: 2rem;
    margin: 2.4rem auto;
  }
}

@media (max-width:768px) {
  .c1 {
    padding-inline: 3.2rem;
  }
}

<div
  class="c0"
  id="teste"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3 c4"
      >
        My grid
      </h2>
      <p
        class="c5"
      >
        Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
      </p>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <h3
          class="c3 c8"
        >
          Teste 1
        </h3>
        <p
          class="c5"
        >
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
        </p>
      </div>
      <div
        class="c7"
      >
        <h3
          class="c3 c8"
        >
          Teste 2
        </h3>
        <p
          class="c5"
        >
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
        </p>
      </div>
      <div
        class="c7"
      >
        <h3
          class="c3 c8"
        >
          Teste 3
        </h3>
        <p
          class="c5"
        >
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
        </p>
      </div>
    </div>
  </div>
</div>
`;
